### PR TITLE
Fix edit timetable flow

### PR DESCRIPTION
**Pull request by majimearun**
- Fixed flow of only allowing draft timetables to be editable (besides metadata)
- Made sure empty timetables cannot be publihed

### CODE REVIEW

File: src/controllers/timetable/editTimetableMetadata.ts [hunk 1]
```diff
@@ -74,6 +74,18 @@ export const editTimetableMetadata = async (req: Request, res: Response) => {
   }
 
   try {
+    const currentState = timetable.private && timetable.draft;
+    const numSections = await timetableRepository
+      .createQueryBuilder("timetable")
+      .innerJoinAndSelect("timetable.sections", "sections")
```
Comment (soumitradev) [inline src/controllers/timetable/editTimetableMetadata.ts:80]: Do this join in the previous query that checks for existence of timetable, and check for number of sections after checking existence

**Review by soumitradev — COMMENTED** [review]
One small change, other than that, LGTM

File: src/controllers/timetable/editTimetableMetadata.ts (hunk 1 above)
Comment (soumitradev) [inline src/controllers/timetable/editTimetableMetadata.ts:80]: Do this join in the previous query that checks for existence of timetable, and check for number of sections after checking existence

**Review by soumitradev — CHANGES_REQUESTED** [review]
One small change, other than that, LGTM

File: src/controllers/timetable/editTimetableMetadata.ts [hunk 2]
```diff
@@ -69,23 +70,24 @@ export const editTimetableMetadata = async (req: Request, res: Response) => {
     return res.status(404).json({ message: "timetable not found" });
   }
 
+  const currentState = timetable.private && timetable.draft;
+  const numSections = timetable.sections.length;
+
+  if (
+    currentState &&
+    numSections == 0 &&
+    (isDraft == false || isPrivate == false)
```
Comment (soumitradev) [inline src/controllers/timetable/editTimetableMetadata.ts:79]: make all of your equality checks type strict

File: src/controllers/timetable/editTimetableMetadata.ts [hunk 3]
```diff
@@ -69,23 +70,24 @@ export const editTimetableMetadata = async (req: Request, res: Response) => {
...
+  ) {
+    return res.status(400).json({
+      message: "Cannot make a timetable public without any sections",
```
Comment (soumitradev) [inline src/controllers/timetable/editTimetableMetadata.ts:82]: first letter of error message is always small

File: src/controllers/timetable/editTimetableMetadata.ts [hunk 4]
```diff
@@ -69,23 +70,24 @@ export const editTimetableMetadata = async (req: Request, res: Response) => {
     return res.status(404).json({ message: "timetable not found" });
   }
 
+  const currentState = timetable.private && timetable.draft;
```
Comment (soumitradev) [inline src/controllers/timetable/editTimetableMetadata.ts:73]: we can simplify this to `timetable.draft` right?

**Review by soumitradev — CHANGES_REQUESTED** [review]
Other than some small fixes, LGTM

File: src/controllers/timetable/editTimetableMetadata.ts (hunk 1 above)
Comment (majimearun) [reply src/controllers/timetable/editTimetableMetadata.ts]: done

File: src/controllers/timetable/editTimetableMetadata.ts (hunk 2 above)
Comment (majimearun) [reply src/controllers/timetable/editTimetableMetadata.ts]: done

File: src/controllers/timetable/editTimetableMetadata.ts (hunk 3 above)
Comment (majimearun) [reply src/controllers/timetable/editTimetableMetadata.ts]: done

File: src/controllers/timetable/editTimetableMetadata.ts (hunk 4 above)
Comment (majimearun) [reply src/controllers/timetable/editTimetableMetadata.ts]: done

**Review by soumitradev — APPROVED** [review]

**Pull request merged**